GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -174,13 +174,20 @@ public virtual bool TryCreateToolsDirectory(out string upgraderToolPath, out str
                     catch (UnauthorizedAccessException e)
                     {
                         success = false;
-                        error = string.Join(Environment.NewLine, "Unauthorized access.", "Please retry gvfs upgrade from an elevated command prompt.", e.ToString());
+                        error = string.Join(
+                            Environment.NewLine, 
+                            "Unauthorized access.", 
+                            $"Make sure you have write permissions to directory {rootDirectoryPath} and run `gvfs upgrade` again.", 
+                            e.ToString());
                         break;
                     }
                     catch (IOException e)
                     {
                         success = false;
-                        error = "Disk error while trying to copy upgrade tools." + Environment.NewLine + e.ToString();
+                        error = string.Join(
+                            Environment.NewLine, 
+                            "Disk error while trying to copy upgrade tools.", 

```

Comment:
Minor: Should we say "I/O error" instead of "Disk error"?

"Disk error" makes me think there was a hardware failure (and I don't think that we can assume that that IOException is for hardware failure).